hack/make-rules/update.sh

```diff
@@ -76,4 +72,4 @@ for t in ${BASH_TARGETS}; do
 	fi
 done
 
-echo -e "${color_green}Update scripts completed successfully${color_norm}"
+echo -e "${color_green:?}Update scripts completed successfully${color_norm}"
```

Comment:
```
In ./hack/make-rules/update.sh line 61:
        echo -e "${color_yellow}Running ${t}${color_norm}"
                 ^-------------^ SC2154: color_yellow is referenced but not assigned.
                                            ^-----------^ SC2154: color_norm is referenced but not assigned.

In ./hack/make-rules/update.sh line 64:
                        echo -e "${color_red}Running ${t} FAILED${color_norm}"
                                 ^----------^ SC2154: color_red is referenced but not assigned.
```

Reply:
These should be coming in from `lib/util.sh`, I would rather we understand why that's not getting sourced in such that these are visible, rather than defend against them being unassigned

Reply:
@spiffxp Thank you so much for the review and comments.
Is it better to add the following comment like #73746 than this modification?

for example
```
# shellcheck disable=SC2154 # color_green defined in lib/util.sh
```

Reply:
Why is shellcheck not seeing this variable come in from that script, is this just not something it's capable of seeing?

Reply:
I think that shellcheck does not detect such variables and says warning from the following document.
- https://github.com/koalaman/shellcheck/wiki/SC2154 -> Exceptions

According to the document, I think that `${var:?}` or disabling is reasonable change.
Should we consider any other changes?

Reply:
>Note: This message only triggers for variables with lowercase characters in their name (foo and kFOO but not FOO) due to the standard convention of using lowercase variable names for unexported, local variables.

Is the explanation I was looking for.  OK, yeah, `${var:?}` seems valid here then.

Reply:
the other "right" way to address this would be to insist all of the `color_foo` vars be upcased, but I'm not going to insist on that as it will blow scope here and likely fall prey to merge conflicts

Reply:
Thanks, I see.